webauthn/authentication/verify_authentication_response.py

```diff
@@ -102,18 +103,10 @@ def verify_authentication_response(
             "Client data challenge was not expected challenge"
         )
 
-    if isinstance(expected_origin, str):
-        if expected_origin != client_data.origin:
-            raise InvalidAuthenticationResponse(
-                f'Unexpected client data origin "{client_data.origin}", expected "{expected_origin}"'
-            )
-    else:
-        try:
-            expected_origin.index(client_data.origin)
-        except ValueError:
-            raise InvalidAuthenticationResponse(
-                f'Unexpected client data origin "{client_data.origin}", expected one of {expected_origin}'
-            )
+    if not validate_expected_origin(expected_origin, client_data.origin):
+        raise InvalidAuthenticationResponse(
+            f'Unexpected client data origin "{client_data.origin}", expected "{expected_origin}"'
+        )
```

Comment:
We've lost some granularity on the error message here - but it's easy to restore if required.